### ISSUE TITLE
Unknown issuer kind: Issuer

### ISSUE THREAD

**Issue by shanehughes1990** [open-telemetry/opentelemetry-helm-charts#1403]
Am trying to deploy the Otel Collector operator, and I had it installed and running in the cluster (in a devops namespace where I do my initial config set for what will go live).

Everything was running great we tested it all that jazz, I tore it down and moved the configs into another namespace, Now the otel operator refuses to start due to missing the opentelemetry-operator-serving-cert secret. The secret is not getting generated due to `Unknown issuer kind: Issuer`

I've checked for lingering resources in my devops namespace and cleaned up a few secrets, etc things that were left behind. (Including the secret that was created in devops namespace that it's now missing)

I'm at a loss of what to do next, as this was all working the only change was tearing it down and moving namespaces (we use TF for all our deployments)

The helm chart values
```terraform
resource "helm_release" "otel_operator" {
  depends_on = [kubernetes_namespace.environment_namespace]
  namespace  = kubernetes_namespace.environment_namespace.metadata[0].name
  repository = "https://open-telemetry.github.io/opentelemetry-helm-charts"
  chart      = "opentelemetry-operator"
  version    = "0.71.1"
  name       = "opentelemetry-operator"

  values = [
    yamlencode({
      "manager" : {
        "collectorImage" : {
          "repository" : "otel/opentelemetry-collector-k8s",
        },
      }
    }),
  ]
}
```

The latest events from the deploy
```sh
13m         Normal    cert-manager.io      certificaterequest/opentelemetry-operator-serving-cert-1   Certificate request has been approved by cert-manager.io
13m         Normal    MissingSecret        certificaterequest/opentelemetry-operator-serving-cert-1   Referenced secret metrics/opentelemetry-operator-serving-cert-ckz2b not found: secrets "opentelemetry-operator-serving-cert-ckz2b" not found
13m         Normal    Issuing              certificate/opentelemetry-operator-serving-cert            Issuing certificate as Secret does not exist
13m         Normal    Generated            certificate/opentelemetry-operator-serving-cert            Stored new private key in temporary Secret resource "opentelemetry-operator-serving-cert-4fsnt"
13m         Normal    Generated            certificate/opentelemetry-operator-serving-cert            Stored new private key in temporary Secret resource "opentelemetry-operator-serving-cert-ckz2b"
13m         Normal    Requested            certificate/opentelemetry-operator-serving-cert            Created new CertificateRequest resource "opentelemetry-operator-serving-cert-1"
13m         Warning   Failed               certificate/opentelemetry-operator-serving-cert            The certificate request has failed to complete and will be retried: Unknown issuer kind: Issuer
13m         Normal    ScalingReplicaSet    deployment/opentelemetry-operator                          Scaled up replica set opentelemetry-operator-6654fd4d4c to 1
```

The container itself, stuck due to waiting for volume mount from the secret
```sh
opentelemetry-operator-6654fd4d4c-4gnqs     0/2     ContainerCreating   0          19m
```

The describe of the pod
```yaml
Name:             opentelemetry-operator-6654fd4d4c-4gnqs
Namespace:        metrics
Priority:         0
Service Account:  opentelemetry-operator
Node:             gke-development-gke-beta-pool-2595f896-dqgm/10.5.250.9
Start Time:       Thu, 31 Oct 2024 09:44:05 -0400
Labels:           app.kubernetes.io/component=controller-manager
                  app.kubernetes.io/name=opentelemetry-operator
                  pod-template-hash=6654fd4d4c
Annotations:      kubectl.kubernetes.io/default-container: manager
Status:           Pending
IP:               
IPs:              <none>
Controlled By:    ReplicaSet/opentelemetry-operator-6654fd4d4c
Containers:
  manager:
    Container ID:  
    Image:         ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator:0.110.0
    Image ID:      
    Ports:         8080/TCP, 9443/TCP
    Host Ports:    0/TCP, 0/TCP
    Command:
      /manager
    Args:
      --metrics-addr=0.0.0.0:8080
      --enable-leader-election
      --health-probe-addr=:8081
      --webhook-port=9443
      --collector-image=otel/opentelemetry-collector-k8s:0.110.0
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     100m
      memory:  128Mi
    Requests:
      cpu:      100m
      memory:   64Mi
    Liveness:   http-get http://:8081/healthz delay=15s timeout=1s period=20s #success=1 #failure=3
    Readiness:  http-get http://:8081/readyz delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment:
      ENABLE_WEBHOOKS:  true
    Mounts:
      /tmp/k8s-webhook-server/serving-certs from cert (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-k4pt8 (ro)
  kube-rbac-proxy:
    Container ID:  
    Image:         quay.io/brancz/kube-rbac-proxy:v0.15.0
    Image ID:      
    Port:          8443/TCP
    Host Port:     0/TCP
    Args:
      --secure-listen-address=0.0.0.0:8443
      --upstream=http://127.0.0.1:8080/
      --logtostderr=true
      --v=0
    State:          Waiting
      Reason:       ContainerCreating
    Ready:          False
    Restart Count:  0
    Limits:
      cpu:     500m
      memory:  128Mi
    Requests:
      cpu:        5m
      memory:     64Mi
    Environment:  <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-k4pt8 (ro)
Conditions:
  Type                        Status
  PodReadyToStartContainers   False 
  Initialized                 True 
  Ready                       False 
  ContainersReady             False 
  PodScheduled                True 
Volumes:
  cert:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  opentelemetry-operator-controller-manager-service-cert
    Optional:    false
  kube-api-access-k4pt8:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason       Age                From               Message
  ----     ------       ----               ----               -------
  Normal   Scheduled    20m                default-scheduler  Successfully assigned metrics/opentelemetry-operator-6654fd4d4c-4gnqs to gke-development-gke-beta-pool-2595f896-dqgm
  Warning  FailedMount  2m (x17 over 20m)  kubelet            MountVolume.SetUp failed for volume "cert" : secret "opentelemetry-operator-controller-manager-service-cert" not found
```

**Comment by KingJ** [comment]
@TylerHelmuth I've also been hitting an issue with Cert Manager Certificates failing to issue with a `Unknown issuer kind: Issuer` error recently, albeit not with a certificate for the OTEL operator.

I've spent some time troubleshooting this with the Cert Manager maintainers this morning, and we've determined this is because of a 3rd party issuer that we have on our clusters - it is handling (and rejecting) `CertificateRequests` that do not set `spec.issuerRef.group`, which should really only be handled by Cert Manager's own internal issuers. [A bug has been raised up against that external issuer](https://github.com/cloudflare/origin-ca-issuer/issues/149) - but this could potentially be caused by other external issuers too.

This only happens when `spec.issuerRef.group` is unset. While it's perfectly valid for this to be unset and effectively defaults to `cert-manager.io`, if the OTEL Operator chart was updated to [explicitly set `group: cert-manager.io` here](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/4a6037437ef8bebb9fab0cfeeaa9f4bb76ffb33f/charts/opentelemetry-operator/templates/certmanager.yaml#L24-L29), then it would work around this issue regardless of the bug fix on the external issuer side.

I spent hours tearing my hair out over this yesterday, and this is pretty much the only other public report of this issue - so hopefully this helps!
